src/fourier.jl

```diff
@@ -0,0 +1,773 @@
+        n_intvs += 1
+    end
+    return (n_ph / (n_intvs * segment_size))
+end
+
+@resumable function get_flux_iterable_from_segments(times:: AbstractVector{<:Real}, 
+                                                    gti::AbstractMatrix{<:Real}, 
+                                                    segment_size::Real; n_bin=nothing,
+                                                    fluxes=nothing, errors=nothing)
+    dt = nothing
+    binned = !isnothing(fluxes)
+    if binned
+        dt = Statistics.median(diff(times[1:100]))
+    end
+    fun = _which_segment_idx_fun(;binned, dt)
+
+    for (s, e, idx0, idx1) in fun(times, gti, segment_size)
+        if idx1 - idx0 < 2
+            @yield nothing
+            continue
+        end
+        if !binned
+            event_times = times[idx0:idx1-1]
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = fit(Histogram,Float64.(event_times .- s);nbins=n_bin).weights
+        else
+            cts = Float64.(fluxes[idx0+1:idx1])
+            if !isnothing(errors)
+                cts = cts, errors[idx0+1:idx1]
+            end
+        end
+        @yield cts
+    end
+end
+
+function avg_pds_from_iterable(flux_iterable, dt::Real; norm::String="frac", 
+                               use_common_mean::Bool=true,
+                               silent::Bool=false)
+    local_show_progress = show_progress
+    if silent
+        local_show_progress = identity
+    end
+
+    # Initialize stuff
+    cross = unnorm_cross = nothing
+    n_ave = 0
+    freq = nothing
+
+    sum_of_photons = 0
+    common_variance = nothing
+    fgt0 = nothing
+    n_bin = nothing
+
+    for flux in local_show_progress(flux_iterable)
+        if isnothing(flux) || all(iszero,flux)
+            continue
+        end
+
+        # If the iterable returns the uncertainty, use it to calculate the
+        # variance.
+        variance = nothing
+        if flux isa Tuple
+            flux, err = flux
+            variance = Statistics.mean(err) ^ 2
+        end
+
+        # Calculate the FFT
+        n_bin = length(flux)
+        ft = fft(flux)
+
+        # This will only be used by the Leahy normalization, so only if
+        # the input light curve is in units of counts/bin
+        n_ph = sum(flux)
+        unnorm_power = real.(ft .* conj.(ft))
+
+        # Accumulate the sum of means and variances, to get the final mean and
+        # variance the end
+        sum_of_photons += n_ph
+
+        if !isnothing(variance)
+            common_variance = sum_if_not_none_or_initialize(common_variance, variance)
+        end
+
+        # In the first loop, define the frequency and the freq. interval > 0
+        if isnothing(cross)
+            fgt0 = positive_fft_bins(n_bin)
+            freq = fftfreq(n_bin, dt)[fgt0]
+        end
+
+        # No need for the negative frequencies
+        unnorm_power = keepat!(unnorm_power,fgt0)
```

Comment:
Am I missing something or `fgt0` here can be also `nothing`?

Reply:
Cross is initialized with `nothing` which means `fgt0 = positive_fft_bins(n_bin)` will always be executed first time.